scipy/stats/_continuous_distns.py

```diff
@@ -3456,8 +3456,25 @@ def _munp(self, n, a, c):
         return sc.poch(a, n*1.0/c)
 
     def _entropy(self, a, c):
-        val = sc.psi(a)
-        return a*(1-val) + 1.0/c*val + sc.gammaln(a) - np.log(abs(c))
+        def regular(a, c):
+            val = sc.psi(a)
+            A = a * (1 - val) + val / c
+            B = sc.gammaln(a) - np.log(abs(c))
+            h = A + B
+            return h
+
+        def asymptotic(a, c):
+            # gammaln(a) ~ a * ln(a) - a - 0.5 * ln(a) +
+            #              0.5 * ln(2 * pi) + 1 / (12 * a)
+            # psi(a) ~ ln(a) - 1 / (2 * a) - 1 / (12 * a ^ 2)
+            P = np.log(a) - 1 / (2 * a) - 1 / (12 * a ** 2)
```

Comment:
This is correct, but 
```suggestion
            P = np.log(a) - 1 / (2 * a) - (a**-2)/12
```
to avoid overflow.